src/pipelinedefinition.cpp

```diff
@@ -349,6 +349,20 @@ class NodeValidator {
         return StatusCode::OK;
     }
 
+    Status checkIfGatherNodeCorrect(const NodeInfo& dependantNodeInfo) const {
```

Comment:
We have validateNodes/validateNode/validateForCycle - maybe follow the convention and call it: validateGatherNode?